### PR TITLE
ip_nexthop,cmdgen: add group extension and code refactor for cmdgen

### PR DESCRIPTION
**Pull request by Ali-aqrabawi**
[+] added group extension to ip-nexthop and group separator 
[+] changed cmdgen logic to generate args for starcmd, this will allow us to change the lyd_node before passing to the args generator usefull for iproute replace action.

### CODE REVIEW

File: src/lib/change_cmdgen.c [hunk 1]
```diff
@@ -308,10 +312,109 @@ int creat_cmd_key_and_value(struct lyd_node *dnode, oper_t curr_op_val, char **k
...
+                                        "but failed to retrieve the arg-name list from ON_UPDATE_INCLUDE extension for node \"%s\" \n",
+                                __func__, next->schema->name);
+                        return NULL;
+                    }
+                    // get all the on_update_include arg-names, fetch them from sr, and add them to the cmd_line.
+                    // args-names format = arg1, arg2, ... argn
+                    char *token;
+
+                    // Get the first arg
+                    token = strtok(on_update_include, ",");
```
Comment (adaraiseh) [inline src/lib/change_cmdgen.c:348]: can we make this a function ?

Comment (adaraiseh) [reply src/lib/change_cmdgen.c:348]: pull from sysrepo:
input: node name(s)
output: either lyd_node or cmd args list

**Review by adaraiseh — CHANGES_REQUESTED** [review]

**Review by vjardin — APPROVED** [review]
some cosmetics, this same commit does many things:
  - adding/changing some features
  - updating some cosmetics
  - fixing some logics
at least, it should deserve 3 separate commits with their respective commit logs.

**Comment by Ali-aqrabawi** [comment]
It the same. Adding support for grouping through ip_nexthop. I can't add the support if the yang yang where grouing is used is not there (ip_nexthop). The code logic is needed for the same support. But added it as details.

**Pull request merged**